react/week-2/lesson.md

```diff
@@ -492,55 +502,66 @@ useEffect(() => {
 And here is a more complete example:
 
 ```js
-import React, { useState, useEffect } from 'react'; // remember to import the Hook(s) you need!
+import React, { useState, useEffect } from "react"; // remember to import the Hook(s) you need!
 
 function MartianPhotoFetcher() {
-  const [marsPhotos, setMarsPhotos] = useState();
+  const [marsPhotos, setMarsPhotos] = useState({});
 
   useEffect(() => {
-    fetch(`https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/photos?earth_date=2015-6-3&api_key=DEMO_KEY`)
+    fetch(
+      `https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/photos?earth_date=2015-6-3&api_key=DEMO_KEY`
+    )
       .then(res => res.json())
       .then(data => setMarsPhotos(data));
-    });
   }, []);
 
   return (
     <div>
-      // TODO: update this example
+      {marsPhotos.photos &&
+        marsPhotos.photos.map((photo, index) => {
+          return (
+            <img
+              key={`mars-photo-${index}`}
+              src={photo.img_src}
+              alt={photo.camera.name}
+            />
+          )
+      })}
     </div>
   );
-};
+}
 
 export default MartianPhotoFetcher;
+
 ```
 
 In the code above, we're saying to React “When this component is mounted, call the NASA photos API, and when you receive a response, save it inside of the 'marsPhotos' state”.
 
 This is a very common pattern which will come in very useful!
 
-| **Exercise** |
+| **Exercise D** |
 | :--- |
 | 1. Open the `pokedex` React application again and open the `src/BestPokemon.js` file. |
-| 2. Create a new component called `BestPokemonFetcher`. |
+| 2. Add a new component inside `src/BestPokemon.js` called `BestPokemonFetcher`. |
 | 3. Change the `export default` to export `BestPokemonFetcher` instead of `BestPokemon`. We **don't** need to make any changes to the `BestPokemon` component. |
-| 4. In the new `BestPokemonFetcher` component, create a new state variable called `pokemon` and initialise it to `null`. <details><summary>Click here if you are stuck.</summary>Look at the State section to see how to create state variables.</details> |
-| 5. If there is no `pokemon` state (hint: `if (!pokemon) {}`), then return `null` so that the component renders nothing. |
-| 6. If there is some `pokemon` state (`else {}`), then render the `BestPokemon` component and pass the `pokemon` state variable as the `pokemon` prop (hint: `<BestPokemon pokemon={pokemon} />`). |
+| 4. In the new `BestPokemonFetcher` component, create a new state variable called `bestPokemon` and initialise it to `null`. <details><summary>Click here if you are stuck.</summary>Look at the State section to see how to create state variables.</details> |
+| 5. If there is no `bestPokemon` state (hint: `if (!bestPokemon) {}`), then return `null` so that the component renders nothing. |
```

Comment:
If stick with the `data && data.map()` pattern I commented on above then we should make this match.

Reply:
I've changed it to `if () {}` and `else {}` in the other example so we can keep this here. But I do think we should show the `&&` and `? :` pattern during class.

Reply:
I've added a section on conditional rendering incl different syntaxes.

Reply:
Thanks, looks good.